File: base/221105/fsrc/common/infos.f90
! INFO_MOD is a module defining exit flags.
!
! Coded by Zaikun ZHANG (www.zhangzk.net).
!
! Started: July 2020.
!
! Last Modified: Sunday, May 22, 2022 PM11:32:18


module infos_mod

use, non_intrinsic :: consts_mod, only : IK

implicit none
private
public :: INFO_DFT
public :: INVALID_INPUT
public :: SMALL_TR_RADIUS
public :: FTARGET_ACHIEVED
public :: TRSUBP_FAILED
public :: MAXFUN_REACHED
public :: MAXTR_REACHED
public :: NAN_INF_X
public :: NAN_INF_F
public :: NAN_MODEL
public :: DAMAGING_ROUNDING
public :: NO_SPACE_BETWEEN_BOUNDS
public :: ZERO_LINEAR_CONSTRAINT


integer(IK), parameter :: INFO_DFT = 0_IK
integer(IK), parameter :: INVALID_INPUT = -100_IK
integer(IK), parameter :: SMALL_TR_RADIUS = 0_IK
integer(IK), parameter :: FTARGET_ACHIEVED = 1_IK
integer(IK), parameter :: TRSUBP_FAILED = 2_IK
integer(IK), parameter :: MAXFUN_REACHED = 3_IK
integer(IK), parameter :: MAXTR_REACHED = 20_IK
integer(IK), parameter :: NAN_INF_X = -1_IK
integer(IK), parameter :: NAN_INF_F = -2_IK
integer(IK), parameter :: NAN_MODEL = -3_IK
integer(IK), parameter :: NO_SPACE_BETWEEN_BOUNDS = 6_IK
integer(IK), parameter :: DAMAGING_ROUNDING = 7_IK
integer(IK), parameter :: ZERO_LINEAR_CONSTRAINT = 8_IK


end module infos_mod

File: .development/archiva/230305/last/matlab/interfaces/+newuoa_last_mat/private/infos.m
% This is a function defining exit flags.
%
% Coded by Zaikun Zhang in July 2020.
%
% Last Modified: Saturday, July 10, 2021 PM10:13:33

function info = infos(name)

    switch name
        case 'invalid_input'
            info = -100;
        case 'small_tr_radius'
            info = 0;
        case 'ftarget_achieved'
            info = 1;
        case 'trsubp_failed'
            info = 2;
        case 'maxfun_reached'
            info = 3;
        case 'maxtr_reached'
            info = 20;
        case 'nan_x'
            info = -1;
        case 'nan_inf_f'
            info = -2;
        case 'nan_model'
            info = -3;
        case 'damaging_rounding'
            info = 7;
        otherwise
            error('Error (info.m): unknown info %s', name);
    end
end
